### DISCUSSION TITLE
Renovate seems to not grab latest github release/tag

### DISCUSSION THREAD

**Discussion by ido-namely** [renovatebot/renovate#16708] [Request Help] [answered]
Hi,
I have a regex manager set to grab a github tag and release from a file.
Renovate opens PRs to update the version, but for some reason does not attempt to update to the latest tag or release.
is there anything I'm missing?

I created an example repo for this issue - https://github.com/ido-namely/renovate-example2

Thanks!

**Comment by rarkins** [comment]
Does this only reproduce if you have all 5 regex managers in config? Or can you simplify this reproduction to be minimal?

**Reply by ido-namely** [reply]
it does, I'll go ahead and simplify it, I just wanted to show it occurs when I use `github-tags` or `github-releases` and when I use different versioningtemplate.

**Reply by ido-namely** [reply]
@rarkins , I removed some of the other regex managers and left one for `grpc/grpc` and one for `grpc/grpc-java`

**Reply by rarkins** [reply]
Does it need both packages to reproduce or can you make it more minimal still?

**Reply by ido-namely** [reply]
sure, now there is only one package/regex manager

**Reply by rarkins** [reply]
Does it need the custom versioning and extractVersion? Shouldn't it work with default semver?

**Reply by ido-namely** [reply]
updated, now using default semver

**Reply by rarkins** [reply]
And can you switch back to tags instead of releases? One less thing which could go wrong

**Reply by ido-namely** [reply]
sure, just changed it.

**Comment by zharinov** [comment]
~~Please try change `extractVersionTemplate` to `^v?(?<version>\\d+\\.\\d+\\.\\d+)` and check the rebase/retry checkbox in the PR body.~~

**Comment by zharinov** [comment, marked as answer]
Found the cause: that's because of `extractVersionTemplate` extracts `v1.48.0` and `v1.48.0-pre1` as the same release, which makes it being considered unstable.

The complete regex is this one: `^v(?<version>\\d+\\.\\d+\\.\\d+(?:-pre\\d+)?)$`

**Reply by ido-namely** [reply]
Thank you @zharinov!
Any suggestions on what option/regex to use in order to avoid Renovate pulling the unstable versions?
I tried adding a package rule:
```
    {
      "matchPackageNames": ["grpc/grpc"],
      "allowedVersions": "/^v\\d+\\.\\d+\\.\\d+$/"
    }
```
but that seemed to have only closed the previous PR.

**Reply by viceice** [reply]
allowed version filter is applied after extract version, so remove the v prefix

**Reply by ido-namely** [reply]
If I remove the package rule and use the regex you've suggested it works, but how would I prevent future cases
where only `v1.48.0-pre1` was released but I would like Renovate to wait for the stable version, i.e `v1.48.0`?

**Reply by ido-namely** [reply]
Thanks @viceice ,
I updated the allowed version regex but now the PR got closed again.
Do you know what I may be missing?

**Reply by zharinov** [reply]
> Any suggestions on what option/regex to use in order to avoid Renovate pulling the unstable versions?

Seems like you don't need to do something, as Renovate should figure out unstable versions for you. Just use regex that captures unstable suffixes which can be produced in the `grpc` project.

**Reply by ido-namely** [reply]
What would Renovate do if v1.48.0 did not exist?

**Reply by rarkins** [reply]
Why can't semver versioning be used to capture the non v part - why is regex even necessary?

**Reply by zharinov** [reply]
> Why can't semver versioning be used to capture the non v part - why is regex even necessary?

Just saw it produced fresh PR without any regex: https://github.com/renovate-testing/test-16708-grpc-github-tags-5/pull/1

**Reply by zharinov** [reply]
But oh I see it places `v` prefix now

**Reply by zharinov** [reply]
> What would Renovate do if v1.48.0 did not exist?

Should wait for `v1.48.0` unless you explicitly choose not to [`ignoreUnstable`](https://docs.renovatebot.com/configuration-options/#ignoreunstable)

**Reply by ido-namely** [reply]
Thanks good to know, thank you.
Would you mind explaining then why the original regex that I have used for `extractVersionTemplate` (in order to remove the `v` prefix) did not work?
`"extractVersionTemplate": "^v(?<version>\\d+\\.\\d+\\.\\d+)$"`

**Reply by zharinov** [reply]
Which one do you mean? The very first was lacking `$` which caused Renovate to mix stable/unstable in weird way. Speaking of `v(?<version>\\d+\\.\\d+\\.\\d+)$`, it's strange to me why it isn't working, need to research more.

**Reply by ido-namely** [reply]
I see, thank you @zharinov . I only now realized it was the missing `$`.
Now it works as expected - https://github.com/ido-namely/renovate-example2/pull/7
